fixes issue 12: EPSG codes incorrect for southern hemisphere

pjhartzell opened the pull request:
- Added a test and test data. As with existing test data, the new test data is a full scene, which is heavy (~10MB).
- Formatters added new lines in several files.

gadomski commented on src/stactools/landsat/mtl_metadata.py:
```diff
@@ -68,10 +68,8 @@ def processing_level(self) -> str:
             utm_zone = self._get_text('PROJECTION_ATTRIBUTES/UTM_ZONE')
-            center_lat = (bbox[1] + bbox[3]) / 2.0
-            return int(f"{326 if center_lat > 0 else 327}{utm_zone}")
+            return int(f"326{utm_zone}")
```
Can you throw a comment in here about how the transform, as provided by USGS, requires the UTM zone to always be north, even if the data are in the southern hemisphere?

pjhartzell replied:
Yep. Done.

gadomski reviewed (changes requested):
Instead of committing these ~100MB files, use the [get external data](https://github.com/stac-utils/stactools/blob/main/src/stactools/testing/test_data.py#L42) functionality from `stactools.testing` to fetch the files for testing.

pjhartzell committed:
fixes issue 12: EPSG codes incorrect for southern hemisphere

- added comment and link to USGS FAQ for this topic
- added test data and appended an assertion check in the existing tests
- format changes (blank lines) were added by yapf

pjhartzell commented:
> Instead of committing these ~100MB files, use the [get external data](https://github.com/stac-utils/stactools/blob/main/src/stactools/testing/test_data.py#L42) functionality from `stactools.testing` to fetch the files for testing.

The original data is about 1 GB, which was a mistake on my part. I replaced those files with resampled data to match the existing test data, which is also resampled. This cuts it down to 6.5 MB, which is in line with the current test data (`assets` subdirectories in `tests/data-files`).

gadomski reviewed (approved):
How does PySTAC do with the v0.7.0 STAC?

pjhartzell commented:
> How does PySTAC do with the v0.7.0 STAC?

OK? There are tests that successfully manipulate Items generated with `Item.from_file` from v0.7.0 STAC json files. The output looks reasonable as well.
